cmd/launcher/doctor.go

```diff
@@ -0,0 +1,541 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/logutil"
+	"github.com/kolide/kit/version"
+	"github.com/kolide/launcher/pkg/agent/flags"
+	"github.com/kolide/launcher/pkg/agent/knapsack"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/autoupdate"
+	"github.com/kolide/launcher/pkg/autoupdate/tuf"
+	"github.com/kolide/launcher/pkg/launcher"
+	"github.com/kolide/launcher/pkg/log/checkpoint"
+	"github.com/peterbourgon/ff/v3"
+
+	"github.com/fatih/color"
+	"golang.org/x/exp/slices"
+)
+
+var (
+	// Command line colors and printing functions
+	cRunning = color.New(color.FgCyan)
+	cHeader  = color.New(color.FgHiWhite).Add(color.Bold)
+	cWarn    = color.New(color.FgHiYellow)
+	pass     = color.New(color.FgGreen).PrintlnFunc()
+	fail     = color.New(color.FgRed).Add(color.Bold).PrintlnFunc()
+
+	// Println functions for checkup details
+	cInfo = color.New(color.FgWhite)
+	info  = func(a ...interface{}) {
+		cInfo.Println(fmt.Sprintf("    %s", a...))
+	}
+	warn = func(a ...interface{}) {
+		cWarn.Println(fmt.Sprintf("    %s", a...))
+	}
+	bad = func(a ...interface{}) {
+		cInfo.Println(fmt.Sprintf("❌  %s", a...))
+	}
+	good = func(a ...interface{}) {
+		cInfo.Println(fmt.Sprintf("✅  %s", a...))
+	}
+
+	configFile string
+)
+
+// checkup encapsulates a launcher health checkup
+type checkup struct {
+	name   string
+	check  func() (string, error)
+	failed bool
+}
+
+func runDoctor(args []string) error {
+	logger := log.With(logutil.NewCLILogger(true), "caller", log.DefaultCaller)
+	opts, err := parseDoctorOptions(os.Args[2:])
+	if err != nil {
+		level.Info(logger).Log("err", err)
+		os.Exit(1)
+	}
+
+	fcOpts := []flags.Option{flags.WithCmdLineOpts(opts)}
+	flagController := flags.NewFlagController(logger, nil, fcOpts...)
+	k := knapsack.New(nil, flagController, nil)
+
+	cRunning.Println("Kolide launcher doctor version:")
+	version.PrintFull()
+	cRunning.Println("\nRunning Kolide launcher checkups...")
+
+	checkups := []*checkup{
```

Comment:
Would we want to also check for --

* Launcher process exists?
* Launcher desktop process exists?
* Is there anything we could check for osquery?

Reply:
I'll add osquery binary and logs checks next.

A launcher process report could be useful info